Fix: tail fails under pseudo with "couldn't allocate absolute path"

naoNao89 opened the pull request:
tail calls .canonicalize() on stdin pipes → triggers realpath() → pseudo fails with "couldn't allocate absolute path"

Fix: Check file type first. Skip canonicalization for pipes, only canonicalize regular files.

Fixes #9292 

Verified: Tested with pseudo in Ubuntu container


```
podman run --rm -v $(pwd):/src:Z -w /src ubuntu:24.04 /bin/bash -c '
  apt update -qq && apt install -y -qq curl pseudo build-essential > /dev/null &&
  curl --proto "=https" --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain stable &&
  . $HOME/.cargo/env &&
  cargo build --release -p uu_tail &&
  echo "/usr/bin/arch
/bin/busybox.nosuid 50" > /tmp/test-data.txt &&
  echo "=== Test WITHOUT pseudo ===" &&
  grep "50\$" /tmp/test-data.txt | ./target/release/tail -n 1 &&
  echo "=== Test WITH pseudo (issue #9292) ===" &&
  mkdir -p /tmp/pseudo && export PSEUDO_PREFIX=/tmp/pseudo &&
  pseudo sh -c "grep \"50\\\$\" /tmp/test-data.txt | ./target/release/tail -n 1" &&
  echo "✓ Fix verified: No '\''couldn'\''t allocate absolute path'\'' errors!"
'
```

```
=== Test WITHOUT pseudo ===
/bin/busybox.nosuid 50
✓ Pass

=== Test WITH pseudo (issue #9292) ===
/bin/busybox.nosuid 50
✓ Fix verified: No 'couldn't allocate absolute path' errors!
```

Ecordonnier commented on src/uu/tail/src/tail.rs:
```diff
@@ -205,6 +205,23 @@ fn tail_stdin(
+    // On non-macOS systems, use is_stdin_directory() which works correctly
+    #[cfg(not(target_os = "macos"))]
+    {
+        if uucore::fs::is_stdin_directory(&stdin()) {
```
@naoNao89 ping, have you seen this comment?

Ecordonnier commented:
Can you please check the CI issue?

naoNao89 committed:
fix(tail): avoid canonicalizing piped stdin

Skip canonicalization for non-regular stdin paths so tail works under pseudo, while preserving directory diagnostics and stabilizing follow-mode tests.

Fixes #9292
